examples/quicksort.jl

```diff
@@ -0,0 +1,53 @@
+"""
+Sorting a 1D CuArray can be done with default `sort!` function, or by invoking
+`CUDA.quicksort!`. `quicksort!` supports an optional maximum of threads per
+kernel. `sort!` will use the maximum allowed value for your hardware. Specifying
+a lower value is helpful if you are running julia on debug level >=2 and kernels
+require more registers.
+"""
+
+using CUDA, Random, Test
+
+# Set to false so that the julia base sort! will fail if that gets invoked.
+CUDA.allowscalar(false)
+
+@test_throws AssertionError quicksort!(CuArray([1, 2, 3]), 3)
+
+# must ask for a number of threads which is a power of 2
+#@test_throws AssertionError quicksort!(CuArray([1, 2, 3]), 3)
+
+function test_large_sort(T, f)
+    full_size = CUDA.quicksort_block_size(T)
+    half_size = full_size ÷ 2
+    N = full_size ^ 2 * 8
+    a = map(T, map(f, 1:N))
+    c = CuArray(a)
+    sort!(c)
+    @test c == sort(a)
+    c = CuArray(a)
+    CUDA.quicksort!(c, half_size)
+    @test c == sort(a)
+end
+
+
+# test pre-sorted
```

Comment:
I don't think these warrant a new `examples` file (which are fairly expensive to test, too). Just put these in `test/sorting`.jl?